fix: Error while starting language server 'blocLanguageServer' (pid=null)

after update bloc plugin to 4.1.0 it depend LSP4IJ and can't sart server, I have activated bloc_tools

`com.redhat.devtools.lsp4ij.server.CannotStartServerException: Error while starting language server 'blocLanguageServer' (pid=null)
	at com.redhat.devtools.lsp4ij.LanguageServerWrapper.lambda$start$15(LanguageServerWrapper.java:404)
	at java.base/java.util.concurrent.CompletableFuture.uniExceptionally(CompletableFuture.java:990)
	at java.base/java.util.concurrent.CompletableFuture$UniExceptionally.tryFire(CompletableFuture.java:974)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1773)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1760)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:507)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1491)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:2073)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:2035)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:187)
Caused by: java.lang.RuntimeException: Exception occurred while creating an instance of the stream connection provider
	at com.redhat.devtools.lsp4ij.server.definition.extension.ExtensionLanguageServerDefinition.createConnectionProvider(ExtensionLanguageServerDefinition.java:57)
	at com.redhat.devtools.lsp4ij.LanguageServerWrapper.lambda$start$4(LanguageServerWrapper.java:290)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768)
	... 6 more
Caused by: java.lang.IllegalStateException: bloc_tools installation failed
	at com.bloc.intellij_generator_plugin.language_server.BlocLanguageServerFactory.createConnectionProvider(BlocLanguageServerFactory.kt:21)
	at com.redhat.devtools.lsp4ij.server.definition.extension.ExtensionLanguageServerDefinition.createConnectionProvider(ExtensionLanguageServerDefinition.java:55)
	... 8 more
`

--- a/extensions/intellij/intellij_generator_plugin/src/main/java/com/bloc/intellij_generator_plugin/language_server/BlocLanguageServer.kt
+++ b/extensions/intellij/intellij_generator_plugin/src/main/java/com/bloc/intellij_generator_plugin/language_server/BlocLanguageServer.kt
@@ -2,8 +2,8 @@ package com.bloc.intellij_generator_plugin.language_server
 
 import com.bloc.intellij_generator_plugin.util.BlocPluginNotification
 import com.bloc.intellij_generator_plugin.util.CommandLineHelper
+import com.bloc.intellij_generator_plugin.util.MultiplatformCommandLine
 import com.bloc.intellij_generator_plugin.util.VersionComparator
-import com.intellij.execution.configurations.GeneralCommandLine
 import com.intellij.notification.NotificationType
 import com.intellij.openapi.diagnostic.Logger
 import com.intellij.openapi.project.Project
@@ -11,7 +11,7 @@ import com.redhat.devtools.lsp4ij.server.OSProcessStreamConnectionProvider
 import java.util.concurrent.atomic.AtomicBoolean
 
 class BlocLanguageServer(private val project: Project? = null) :
-    OSProcessStreamConnectionProvider(GeneralCommandLine("bloc", "language-server")) {
+    OSProcessStreamConnectionProvider(MultiplatformCommandLine("bloc", "language-server")) {
 
     companion object {
         private const val PROCESS_TIMEOUT_SECONDS: Long = 5
@@ -79,7 +79,7 @@ class BlocLanguageServer(private val project: Project? = null) :
         val errorMessage = "Unable to $verb bloc_tools"
         try {
             val commandLine =
-                GeneralCommandLine("dart", "pub", "global", "activate", "bloc_tools", TARGET_BLOC_TOOLS_VERSION)
+                MultiplatformCommandLine("dart", "pub", "global", "activate", "bloc_tools", TARGET_BLOC_TOOLS_VERSION)
             val process = commandLine.createProcess()
             val exitCode = process.waitFor()
             if (exitCode != 0) {

--- a/extensions/intellij/intellij_generator_plugin/src/main/java/com/bloc/intellij_generator_plugin/util/CommandLineHelper.kt
+++ b/extensions/intellij/intellij_generator_plugin/src/main/java/com/bloc/intellij_generator_plugin/util/CommandLineHelper.kt
@@ -1,21 +1,20 @@
 package com.bloc.intellij_generator_plugin.util
 
-import com.intellij.execution.configurations.GeneralCommandLine
 import com.intellij.openapi.diagnostic.Logger
 import java.util.concurrent.TimeUnit
 
 object CommandLineHelper {
     private val logger = Logger.getInstance(CommandLineHelper::class.java)
 
-    fun executeWithTimeout(timeout: Long, vararg command: String): String? {
+    fun executeWithTimeout(timeout: Long, exePath: String, vararg parameters: String): String? {
         try {
-            val commandLine = GeneralCommandLine(*command)
+            val commandLine = MultiplatformCommandLine(exePath, *parameters)
             val process = commandLine.createProcess()
             val hasExited = process.waitFor(timeout, TimeUnit.SECONDS)
             if (!hasExited || process.exitValue() != 0) return null
             return process.inputStream.bufferedReader().readLine().orEmpty().trim()
         } catch (e: Exception) {
-            logger.debug("fail to execute command (${command.joinToString(" ")}): ${e.message}")
+            logger.debug("fail to execute command ($exePath ${parameters.joinToString(" ")}): ${e.message}")
             return null
         }
     }

--- a/extensions/intellij/intellij_generator_plugin/src/main/java/com/bloc/intellij_generator_plugin/util/MultiplatformCommandLine.kt
+++ b/extensions/intellij/intellij_generator_plugin/src/main/java/com/bloc/intellij_generator_plugin/util/MultiplatformCommandLine.kt
@@ -1,0 +1,13 @@
+package com.bloc.intellij_generator_plugin.util
+
+import com.intellij.execution.configurations.GeneralCommandLine
+import com.intellij.openapi.util.SystemInfo
+
+class MultiplatformCommandLine(exePath: String, vararg parameters: String): GeneralCommandLine() {
+    init {
+        val adjustedExePath =
+            if (SystemInfo.isWindows && !exePath.endsWith(".bat", ignoreCase = true)) "$exePath.bat" else exePath
+        setExePath(adjustedExePath)
+        addParameters(*parameters)
+    }
+}

